addons/web/static/src/views/fields/translation_button.js

```diff
@@ -50,7 +50,7 @@ export function useTranslationDialog() {
             resModel: resModel,
             userLanguageValue: record.data[fieldName] || "",
             isComingFromTranslationAlert: false,
-            updateField,
+            record: record,
```

Comment:
suggestion: give a `onSave` callback in props instead of the record (to avoid leaking implem details from a component to another):
```js
async onSave() {
    await record.load({}, { keepChanges: true });
    record.model.notify();
}
``` 

and call `this.props.onSave();` where you currently reload the record in the dialog.